### ISSUE TITLE
We don't just prefer, but require the use of Postgres

### ISSUE THREAD

**Issue by nickcharlton** [thoughtbot/suspenders#1245]
I've been working on [updating a little script I have for testing out Administrate](https://github.com/nickcharlton/dotfiles/blob/main/bin/make-administrate-demo), where I'll go about creating a bunch of small Rails applications for testing. This has one problem of leaving a lot of unused, potentially clashing Postgres databases sitting around.

To work around this problem, I considered using SQLite — [especially as there's been significant renewed interest in it](https://fractaledmind.github.io/tags/sqlite/) up to and including Rails 8.0.

Whilst [we prefer Postgres](https://github.com/thoughtbot/dotfiles/pull/728) (and I don't see a need to generally diverge from this opinion), the way we've implemented `lib/install/web.rb` stops us from using another database type, for example:

```sh
$ rails new app_name \
 --suspenders-local \
 --skip-rubocop \
 --skip-test \
 -d=postgresql \
 -m=~/projects/thoughtbot/suspenders/lib/install/web.rb

[… snip …]

remove  config/initializers/cors.rb
      remove  config/initializers/new_framework_defaults_8_0.rb
       apply  /Users/nickcharlton/projects/thoughtbot/suspenders/lib/install/web.rb
  in the local project!
The template [/Users/nickcharlton/projects/thoughtbot/suspenders/lib/install/web.rb] could not be loaded. Error:

=== Please use the correct options ===

# Use the latest suspenders release:
rails new <app_name> \
--skip-rubocop \
--skip-test \
-d=postgresql \
-m=https://raw.githubusercontent.com/thoughtbot/suspenders/main/lib/install/web.rb

# OR use the current (possibly unreleased) `main` branch of suspenders:
rails new <app_name> \
--suspenders-main \
--skip-rubocop \
--skip-test \
-d=postgresql \
-m=https://raw.githubusercontent.com/thoughtbot/suspenders/main/lib/install/web.rb
```

The implementation of `lib/install/web.rb` looks like this:  https://github.com/thoughtbot/suspenders/blob/a4a4af047526c3f4f79b0bed40c7610b185789da/lib/install/web.rb#L25-L33

Can anyone think of a better way to do that test in `options[:database]`, etc? The intention is to see if we should setup and run our suspenders generator, but it's doing more than that.

**Comment by stevepolitodesign** [comment]
@nickcharlton thank you for bringing this up. We're currently working on an internal project to help us re-evaluate our assumptions when it comes to building new Rails 8 applications, and choosing a database is one of the issues I'm most interested in.

**Comment by nickcharlton** [comment]
Yeah, I think it's going to be interesting to see what we might like to change.

I think in this particular situation, we can be less strict about what we allow to use suspenders, but still push towards Postgres. What do you think?

**Comment by stevepolitodesign** [comment]
> I think in this particular situation, we can be less strict about what we allow to use suspenders, but still push towards Postgres.

I agree.

**Comment by nickcharlton** [comment]
I was thinking a lot about how to loosen the check instead of simply adding SQLite, but I couldn't come up with a way of doing it. Instead, I started down the path of having a "supported" database, which inspired how #1248 is working.
